temba/mixins.py

```diff
@@ -0,0 +1,52 @@
+from django.shortcuts import redirect
+
+
+class NotFoundRedirectMixin:
+    """
+    Mixture for redirection when model object can't be found.
+
+    :param redirect_checking_model: a model that is using for filtering, by default it's a model of view.
+    :param redirect_params: dict with configuratuions
+
+        :param filter_key: keyword that is using for model filtering, by default is same as `filter_value`
+        :param filter_value: keyword for selecting value from the request kwargs to be used for model filtering
+        :param model_manager: using to select exact manager of model
+        :param redirect_context: dict of values that will be passed to the session for processing
+        in the redirected view,
+
+    :param redirect_url: an url for redict, when object not found
+
+    """
+    redirect_checking_model = None
+    redirect_params = {
+        "filter_key": None,
+        "filter_value": 'pk',
+        "model_manager": "objects",
+        "redirect_context": None,
```

Comment:
Don't need this `redirect_context` if you have `messages` from `django.contrib` to display the redirect message